src/www/ui/api/Middlewares/RestAuthMiddleware.php

```diff
@@ -43,11 +43,14 @@ public function __invoke(Request $request, RequestHandler $handler) : ResponseIn
   {
     $requestUri = $request->getUri();
     $requestPath = strtolower($requestUri->getPath());
-    $authFreePaths = ["/version", "/info", "/openapi", "/health"];
+    $authFreePaths = ["/info", "/openapi", "/health"];
 
     $isPassThroughPath = false;
+    // path is /repo/api/v2/<endpoint>, we need to get only the endpoint part
+    $parts = explode("/", $requestPath, 5);
+    $endpoint = "/".end($parts);
```

Comment:
If someone have ditched the `/repo` path from Apache conf, then the `$requestPath` will be `/api/v2/<endpoint>`. Here, `$endpoint = "/".end($parts);` can result in something unexpected.

```php
preg_match("/\/?(?:(?:repo)?\/api\/v\d)(.*)/i", $requestPath, $matches);
$endpoint = "";
if (!empty($matches[1])) {
  $endpoint = $matches[1];
}
```

This will help keep the `/repo` optional and capture anything after `/v<digit>` in the `$endpoint`.

Reply:
However, with a quick check, the proposed solution will still work of `/info`, `/openapi` and `health` endpoints.
But for endpoints like `/api/v2/uploads/{id}/item/{itemId}/info`, the variable endpoint will contain `/{id}/item/{itemId}/info`. Which is wrong but sufficient for the usecase.

Reply:
What do you think @alpianon and @shaheemazmalmmd ?

Reply:
@GMishx as you correctly pointed out, the `/repo` part of the path could be changed in Apache conf, and actually one may remove it or change it to any other thing (eg. `/fossology`).

I guess the regex solution (without the `(?:repo)`) would be the cleanest option. However, I noticed that all the code in the API implementation assumes a path starting with `/repo`, see `src/www/ui/api/index.php`:

```php
/ Extracts the version from the URL
function getVersionFromUri ($uri)
{
  $matches = [];
  preg_match('/\/repo\/api\/v(\d+)/', $uri, $matches);
  return isset($matches[1]) ? intval($matches[1]) : null;
}

// Determine the API version based on the URL
$requestedVersion = isset($_SERVER['REQUEST_URI']) ? getVersionFromUri($_SERVER['REQUEST_URI']) : null;
$apiVersion = in_array($requestedVersion, [ApiVersion::V1, ApiVersion::V2]) ? $requestedVersion : ApiVersion::V1; // Default to "1"

// Construct the base path
$BASE_PATH = "/repo/api/v" .$apiVersion;
```

So either this gets solved globally by expanding the scope of this PR, or IMHO this specific issue may be temporarily solved by using explode+end, and another issue may be opened to globally handle API calls with paths not starting with `/repo` because of a non-standard Apache conf.

Reply:
BTW if you fear that regex would be too slow, this might be an alternative solution that would produce the same result (even if I just assume but I've not tested if it's actually faster):

```php
$pos = strpos($requestPath, "/api/v") + 6;
$pos = strpos($requestPath, "/", $pos);
$endpoint = substr($requestPath, $pos);
```